### DISCUSSION TITLE
Check for new versions on Linux?

### DISCUSSION THREAD

**Discussion by nbolton** [deskflow/deskflow#7644] [Ideas]
Currently, the 'Would you like to check for updates?' dialog displays on every OS, even Linux.

![image](https://github.com/user-attachments/assets/7f1ee71b-e69f-45ad-990e-1a48a6658101)

This certainly has a use case on Windows and macOS, but what about Linux?

This _may_ have some value on Linux, in that sometimes Linux packagers can take a while to update, e.g. the stable version of Debian. If you answer 'Check for updates' you'll know how far behind the latest version you are, but this might not be a concern for someone using a stable distro. If you want to keep up with latest then you'd probably use a rolling distro like Arch or OpenSUSE. In other words, stable distro users don't really care about using the latest versions, and often don't want to.

So, the question is: Should we disable the update check pop (of course, update check is disabled by default) for Linux and only allow it to be enabled via settings?

**Comment by sithlord48** [comment]
Linux has package management this is always redundant.  As a setting its not going to make sense for most users either they won't be able to use the link. As they will still have to update via the package manager when the distro is updates. If anything i would just have in place of a setting and auto update check a  link to the releases page.

**Reply by nbolton** [reply]
I agree that when we make our way into package managers, update check will be completely redundant on Linux because the package version is out of the user's control. For now though, until we're in the repros on Ubuntu, Debian, Fedora, and Arch, perhaps we should keep the update check since people have to install the packages manually... thoughts?

**Reply by sithlord48** [reply]
1.17 or 1.18 could make it into a repo in time we don't know that it will not. So using this to justify . We are on arch btw packages for both a stable and -git were made already https://aur.archlinux.org/packages?O=0&K=deskflow

**Reply by nbolton** [reply]
Good point, so let's disable by default on Linux. I think we should still have a way (in future) to turn it on, but for now there is no such UI element in settings (the only chance you get is the pop on first start).
